test/tests_basic.jl

```diff
@@ -74,6 +102,89 @@ end
     @test_nowarn p4est_destroy(p4est)
     @test_nowarn p4est_connectivity_destroy(connectivity)
   end
+
+  @testset "p8est_connectivity_new_unitcube" begin
+    connectivity = @test_nowarn p8est_connectivity_new_unitcube()
+    @test_nowarn p8est_connectivity_destroy(connectivity)
+  end
+
+  @testset "p4est_refine and p4est_coarsen" begin
+    connectivity = @test_nowarn p4est_connectivity_new_periodic()
+    p4est = @test_nowarn p4est_new(MPI.COMM_WORLD, connectivity, 0, C_NULL, C_NULL)
+
+    refine_fn_c = @cfunction(refine_fn, Cint, (Ptr{p4est_t}, Ptr{p4est_topidx_t}, Ptr{p4est_quadrant_t}))
+    @test_nowarn p4est_refine(p4est, true, refine_fn_c, C_NULL)
+    @test unsafe_load(p4est).global_num_quadrants == 16
+    coarsen_fn_c = @cfunction(coarsen_fn, Cint, (Ptr{p4est_t}, Ptr{p4est_topidx_t}, Ptr{p4est_quadrant_t}))
+    @test_nowarn p4est_coarsen(p4est, true, coarsen_fn_c, C_NULL)
+    @info unsafe_load(p4est).global_num_quadrants
```

Comment:
@ranocha This info gives inconsistent results. I would except it to be 1.

Reply:
See https://github.com/trixi-framework/P4est.jl/pull/55#discussion_r1052987599